### DISCUSSION TITLE
DDR AI characters picking up objects

### DISCUSSION THREAD

**Discussion by FrozenEmpire** [ChilliHugger/The-Lords-Of-Midnight#194] [Code Algorithm Questions] [answered]
Hi Chris,

Am I understanding correct that AI characters can pick up any unique object that they come across, even if it isn't their desired object, and may not be the usual object for their race type (eg Giant could pick up a Bow, etc) ?
At least that's what I have going on :)

Also I think we established that AI characters cannot USE their desired object if they have it, since thats only a UI command for player characters.

(Edit: Forgot that we touched on the above here https://github.com/ChilliHugger/The-Lords-Of-Midnight/discussions/165#discussion-5071230

Also it seems that an AI character drops their unique object (even if they have their desired one), in favour of anything new they come across... I guess thats as per original, but would make more sense that they keep hold of their desired object instead.

Regards
Andrew

**Comment by IcemarkUK** [comment, marked as answer]
Yes - AI characters pick up anything
Yes - AI characters do not use their object
No - AI Characters will never drop an object unless they die. However the will continue looking for their own object even though they will never be able to pick it up. I've considered a gameplay rule to change this...

**Comment by FrozenEmpire** [comment]
I've been using an older offline copy of the codebase which I think did allow AI characters to pick up anything they would find, but see that's not the case in the master.

Definitely seems like a glitch in the original if they can't ever drop the 1st thing they find, yet still search their desired object, and then can't pick it up when they find it.

I'm tempted to modify this in my default rules, to allow them to pickup anything as long as they don't yet have their desired object, and when they do find that, they keep hold of it until they die :)

**Reply by IcemarkUK** [reply]
> I'm tempted to modify this in my default rules, to allow them to pickup anything as long as they don't yet have their desired object, and when they do find that, they keep hold of it until they die :)

I think I'm going to update it as a bug and fix it. Even though it's original, it doesn't help gameplay to leave it there.
